ServerCore/Helpers/LiveEventHelper.cs

```diff
@@ -0,0 +1,240 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using ServerCore.DataModel;
+using ServerCore.ServerMessages;
+
+namespace ServerCore.Helpers
+{
+    public static class LiveEventHelper
+    {
+        private static IServiceProvider serviceProvider;
+
+        public static async Task TriggerNotifications(PuzzleServerContext context, Event currentEvent, int timerWindow, IHubContext<ServerMessageHub> hubContext)
+        {
+            await PreEventReminder(context, currentEvent, timerWindow, hubContext);
+            await EventOpeningReminder(context, currentEvent, hubContext);
+            await EventClosingReminder(context, currentEvent, hubContext);
+        }
+
+        /// <summary>
+        /// Remind teams to go to a scheduled live event
+        /// Notifies all teams that are within the current timer window
+        /// </summary>
+        public static async Task PreEventReminder(PuzzleServerContext context, Event currentEvent, int lengthOfTimerWindow, IHubContext<ServerMessageHub> hubContext)
+        {
+            var liveEvents = await GetLiveEventsForEvent(context, currentEvent, true, false);
+
+            foreach (LiveEvent liveEvent in liveEvents)
+            {
+                string title = $"{liveEvent.Name} starts soon!";
+                string firstReminderMessage = $"Your team's scheduled time for {liveEvent.Name} is in {liveEvent.FirstReminderOffset} minutes!";
+                string lastReminderMessage = $"Your team's scheduled time for {liveEvent.Name} is in {liveEvent.LastReminderOffset} minutes! Time to head to {liveEvent.Location}";
+
+                // Find first reminders
+                var schedule = context.LiveEventsSchedule.Where(s =>
+                    (s.LiveEvent == liveEvent) &&
+                    (s.StartTimeUtc - DateTime.UtcNow) < liveEvent.FirstReminderOffset &&
+                    (s.LastNotifiedUtc - DateTime.UtcNow) > liveEvent.FirstReminderOffset);
+
+                foreach (var teamSlot in schedule)
+                {
+                    await SendNotification(hubContext, title, firstReminderMessage, liveEvent.AssociatedPuzzle.CustomURL, team: teamSlot.Team);
```

Comment:
$"/{this.Event.EventID}/play/Submissions/{puzzleId}" here and everywhere that CustomURL appears